File: ExcAlgoritmoProgramacao_JS/Lista_Bee/1216_ad_hoc.js
import { question } from "readline-sync";

function main() {
    let nomeAmigo = get_Text();
    let distanciaTotal = 0, totalAmigos = 0;

    while (nomeAmigo !== "") {
        totalAmigos++;
        let distanciaCasa = get_Number();
        distanciaTotal += distanciaCasa;
        nomeAmigo = get_Text();
    }

    let distanciaMedia = (distanciaTotal / totalAmigos);
    print(`${distanciaMedia.toFixed(1)}`);
}

function get_Number() {
    return parseInt(question());
}

function get_Text() {
    return question();
}

function print(texto) {
    return console.log(texto);
}

main();

File: ExcAlgoritmoProgramacao_JS/Lista_Bee/1216_ad_hoc.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

int main() {
    char nomeAmigo[100];
    int totalAmigos = 0;
    int distanciaTotal = 0;
    int distanciaCasa;

    while (1) {
        if (!fgets(nomeAmigo, sizeof(nomeAmigo), stdin) || nomeAmigo[0] == '\n') break;
        if (!fgets(nomeAmigo, sizeof(nomeAmigo), stdin) || nomeAmigo[0] == '\n') break;

        if (sscanf(nomeAmigo, "%d", &distanciaCasa) != 1) continue;

        totalAmigos++;
        distanciaTotal += distanciaCasa;
    }

    if (totalAmigos > 0) {
        double distanciaMedia = (double)distanciaTotal / totalAmigos;
        printf("%.1f\n", distanciaMedia);
    }

    return 0;
}